libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs

```diff
@@ -93,7 +88,7 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             };
 
             // Send initial prompt
-            await OnPromptAsync(dc.Context, (IDictionary<string, object>)state[PersistedState], (PromptOptions)state[PersistedOptions], false, cancellationToken).ConfigureAwait(false);
+            await OnPromptAsync(dc.Context, state.MapValueTo<IDictionary<string, object>>(PersistedState), state.MapValueTo<PromptOptions>(PersistedOptions), false, cancellationToken).ConfigureAwait(false);
```

Comment:
We probably don't need to call MapTo on PersistedState and PersistedOptions, persistedState is created a few lines above and promptOptions is passed in as an input parameter